src/sentry/monitors/clock_dispatch.py

```diff
@@ -85,6 +98,91 @@ def _make_reference_ts(ts: datetime):
     return int(ts.replace(second=0, microsecond=0).timestamp())
 
 
+def _evaluate_tick_decision(tick: datetime):
+    """
+    When the clock is ticking, we may decide this tick is invalid and should
+    result in uknown misses and marking all in-progress check-ins as having an
+    unknown result.
+
+    We do this by looking at the historic volume of check-ins for the
+    particular minute boundary we just crossed.
+
+    XXX(epurkhiser): This is currently in development and no decision is made
+    to mark unknowns, instead we are only recording metrics for each clock tick
+    """
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+
+    def get_volume_at_ts(ts: datetime) -> int | None:
+        tick_ts = _make_reference_ts(ts)
+        return _int_or_none(redis_client.get(MONITOR_VOLUME_HISTORY.format(tick_ts)))
+
+    # The clock has just ticked to the next minute. Look at the previous minute
+    # volume across the last with the values
+    past_ts = tick - timedelta(minutes=1)
+    start_tick = past_ts - MONITOR_VOLUME_RETENTION
+
+    # How many check-ins did we have for the tick we just passed?
+    past_minute_volume = get_volume_at_ts(past_ts)
+
+    # Fetch historic volume for this particular minute
+    historic_volume: list[int] = []
+    historic_ts = past_ts
+
+    while historic_ts > start_tick:
+        historic_ts = historic_ts - MONITOR_VOLUME_DECISION_STEP
+        tick_volume = get_volume_at_ts(historic_ts)
+        if tick_volume is not None and tick_volume > 0:
+            historic_volume.append(tick_volume)
+
+    # Can't make any decisions if we didn't have data for the past minute
+    if past_minute_volume is None:
+        return
+
+    # Can't make any decisions if we don't have historic data
+    if len(historic_volume) == 0:
+        return
+
+    # Record some statistics about the past_minute_volume volume in comparison
+    # to the historic_volume data
+
+    historic_mean = statistics.mean(historic_volume)
+    historic_stdev = statistics.stdev(historic_volume)
+
+    historic_stdev_pct = (historic_stdev / historic_mean) * 100
+
+    # Calculate the z-score of our past minutes volume in comparison to the
+    # historic volume data. The z-score is measured in terms of standard
+    # deviations from the mean
```

Comment:
This interpretation of z-score measuring number of standard deviations from the mean can be relied on only if the underlying data is normally distributed. I would recommend looking at the distribution of per-minute volume. If it not normally distributed then I would recommend using different metric. Seer uses [interquartile range](https://medium.com/@divyansh9144/iqr-interquartile-range-for-anomaly-detection-f9c568d1195f) for this same reason.

Reply:
Going to include it for now. I'll take a look out our existing data but I am pretty sure it's going to be relatively normally distributed.